test_unstructured/metrics/test_evaluate.py

```diff
@@ -56,15 +56,20 @@
 
 @pytest.fixture()
 def _cleanup_after_test():
-    # This is where the test runs
+    """Fixture for removing side-effects of running tests in this file."""
+    # Run test as normal
     yield
 
-    os.path.join(TESTING_FILE_DIR, UNSTRUCTURED_OUTPUT_DIRNAME)
-    export_dir = os.path.join(TESTING_FILE_DIR, "test_evaluate_results_cct")
+    def remove_generated_directories():
+        """Remove directories created from running tests"""
+        # Directories to be removed:
+        target_dir_names = ["test_evaluate_results_cct", "test_evaluate_results_cct_txt"]
+        subdirs = (d for d in os.scandir(TESTING_FILE_DIR) if d.is_dir())
+        for d in subdirs:
+            if d.name in target_dir_names:
+                shutil.rmtree(d.path)
```

Comment:
Move this up to the top of the function, line 60 (still needs the blank line before I believe, then keep the yield and the call to this function together. Separating those two by a blank line is expressive I think, but that's the core logic, yield (run the test) and then remove the directories. The definition of the `remove_generated_directories()` function is just defined nested here for convenience and somewhat lower complexity (only used here). The core code should look the same as if that function was defined elsewhere, like at the module level as a peer to this.